src/components/portrait/portrait.component.js

```diff
@@ -6,62 +6,70 @@ import PortraitGravatar from './portrait-gravatar.component';
 import PortraitInitials from './portrait-initials.component';
 import StyledPortrait, { StyledCustomImg, StyledIcon } from './portrait.style';
 
-function Icon(iconProps) {
-  const { darkBackground, shape, size } = iconProps;
-  return (
-    <StyledIcon
-      type='individual'
-      size={ size }
-      shape={ shape }
-      darkBackground={ darkBackground }
-    />
-  );
-}
-
-function Initials(initialsProps) {
-  const {
-    alt, darkBackground, initials, size
-  } = initialsProps;
-  return (
-    <PortraitInitials
-      size={ size }
-      initials={ initials }
-      darkBackground={ darkBackground }
-      alt={ alt }
-    />
-  );
-}
-
-function Gravatar(gravatarProps) {
-  const { alt, gravatar, size } = gravatarProps;
-  return (
-    <PortraitGravatar
-      gravatarEmail={ gravatar }
-      size={ size }
-      alt={ alt }
-    />
-  );
-}
-
-function CustomImg(customimgProps) {
-  const { alt, size, src } = customimgProps;
-  return (
-    <StyledCustomImg
-      src={ src }
-      alt={ alt }
-      size={ size }
-      data-element='user-image'
-    />
-  );
-}
-
 function Portrait(props) {
-  const { gravatar, initials, src } = props;
+  const {
+    alt, darkBackground, gravatar, initials, shape, size, src
+  } = props;
   const showIcon = Boolean(!src && !initials);
   const showInitials = Boolean(!src && initials);
   const showGravatar = Boolean(gravatar);
   const showCustomImg = Boolean(!gravatar && src);
 
+  function renderIcon() {
+    if (!showIcon) {
+      return null;
+    }
+    return (
+      <StyledIcon
+        type='individual'
+        size={ size }
+        shape={ shape }
+        darkBackground={ darkBackground }
+      />
+    );
+  }
+
+  function renderInitials() {
+    if (!showInitials) {
```

Comment:
if (!initials)

Reply:
What is the issue here exactly? The code works, and as it is currently it's easier to see the logic for the component as a whole at a glance, because it's grouped together nicely.

I might suggest inverting the condition so that it's not negated, to read a little better, e.g. 

```
function renderInitials() {
  if (showInitials) {
    /** render component */
  }

  return null;
}
```

But I don't think there's any problem with grouping all of the logic to determine what is and isn't being shown in one place?

Reply:
Ok, I agree, but there is no need for these constants to live outside of the scope of functions that are using them.
```
function renderInitials() {
  const showInitials = Boolean(!src && initials)

  if (showInitials) {
    /** render component */
  }

  return null;
}
```

Reply:
Ok, as discussed off-line, I think either approach is fine, @kyanmichael's initial approach benefitted readability at a glance, but I'm not overly fussed either way so if you feel strongly about it then we'll go with that, as I do see your point! 👍

Reply:
@grabkowski @stevebutler2210: Thanks for your comments/discussion. I've now refactored the code in 478c503 :thumbsup: